languages/csharp/exercises/concept/structs/.docs/after.md

```diff
@@ -0,0 +1,89 @@
+C# `struct`s are closely related `class`s. They have state and behavior. They can have the same kinds of members: constructors, methods, fields, properties, etc.
+
+Fields and properties can be simple types, `struct`s and reference types. `struct`s observe the same rules about scope, read/write rules and access levels as do `class`s.
+
+```csharp
+enum Unit
+{
+    Kg,
+    Lb
+}
+struct Weight
+{
+    private double count;
+    private Unit unit;
+
+    public Weight(double count, Unit unit)
+    {
+        this.count = count;
+        this.unit = unit;
+    }
+
+    public override string ToString()
+    {
+        return count.ToString() + unit.ToString();
+    }
+}
+
+new Weight(77.5, Unit.Kg).ToString();
+// => "77.6Kg"
+```
+
+One of the main things to remember is that when one struct is assigned to a variable or passed as a parameter the values are copied across so changes to he original variable will not affect the copied one and vice versa. In summary, `struct`s are **value types**.
+
+This [article][class-or-struct] discusses the differences between `struct`s and `class`s. You will see from the article that `struct`s tend to be lightweight and [immutable][structs-immutable] although this guidance is not enforced (by default) by the compiler or runtime.
+
+There are a couple of things that you will come up against (and about which the compiler will remind you):
+
+1. Members of a `struct` cannot be initialized inline.
+2. A `struct` cannot be inherited
+
+As a result of point 1 above there is no way for the developer of a `struct` to prevent invalid instances from coming into existence.
+
+#### Common structs
+
+You will see from the documentation that there is a close relationship between primitives and structs. See [`Int32/int]`[int32], for an example. A more conventional example of a`struct`is the type [`TimeSpan`][time-span].
+
+Instances of `TimeSpan` behave much like numbers with comparison operators `>` and `<` and arithmetic operators but unlike equality testing using `Equals()` method (see below) there is no free-ride with the operators. When you implement your own structs you have to [implement][operators] the operators yourself if you want them.
+
+One thing to note about `TimeSpan` is that it implements an interface. Although `struct`s cannot be derived from other `struct`s they can implement interfaces.
+
+#### Equality
+
+Equality testing for `struct`s can often be much simpler than that for `class`s as it simply compares fields for equality by default. There is no need to override `object.Equals()` (or `GetHashCode()`). Remember that if you are relying on `Object.GetHashCode()` you must still ensure that the fields involved in generating the hash code (i.e. all the fields) must not change while a hashed collection is use. Effectively, this means that structs used in this way should be immutable. See (cross-ref-tba).
```

Comment:
You could consider changing `(cross-ref-tba)` to `(TODO cross-ref-tba)` as that might make finding things to do easier. Feel free to ignore.

Reply:
Done